### DISCUSSION TITLE
Questions on schema.

### DISCUSSION THREAD

**Discussion by Jaikant** [beckn/protocol-specifications#199] [Q&A]
I am the cofounder of an RPA based startup focussing on the e-commerce domain.

I was checking out beckn, to see how easy/difficult it is to integrate beckn protocol into our stack, so that took me to the [core schema](https://github.com/beckn/protocol-specifications/blob/master/core/v0/api/core.yaml). 

I really liked the simplicity of the beckn schema. 

Our e-commerce product schema is more complex, and it would be great if someone can share their insights on the beckn schema. So it helps me resolve the conflicts I seem to have between both.

- Why does an `item` have a `parent_item`?
- For Fulfilment, can the schema support multiple vehicles, multiple starts and multiple ends?
- Can we cancel a single item in an oder? The cancel schema only has the order_id.
- Many of the schema description says -  `allows for domain_extension` what does this mean?

Thanks very much for the awesome work and great vision!

**Comment by Jaikant** [comment]
Out of the 4 questions - I guess the 4th one is clear to me. For the first one:
- `Item` with `parent_item` seems like a good way to show add on items. 

The 2 & 3 question would be good to discuss.

**Comment by ravi-prakash-v** [comment]
Item has a `parent_item_id` to allow grouping of items as part of combo offerings.

`AddOn` was separated from `Items` to allow promotional add-ons to be added as part of an entire order and not necessarily in combination with an `Item`.

To cancel a single item in an order, the BAP can call `update` the updated items in the order. Alternatively, if a BPP runs out of stock in an active order, it can call an unsolicited `on_update` with the updated order. 

Regarding the last point, the `allows for domain extension` clause will be removed as the spec is covered under a "Non Derivative" license. That is an error in the spec that will be fixed soon.

**Comment by ravi-prakash-v** [comment]
One suggestion regarding the structure of the thread. It would be great if you can split your questions into a separate thread so that its cleaner and easily searchable.

**Reply by Jaikant** [reply]
@core-wg-admin Okay would do. It has been a while, since I asked these questions and since then I have got some some clarity on some and some are no longer pressing.
